app/lib/audit/backend.dart

```diff
@@ -136,4 +139,56 @@ class AuditBackend {
     InvalidInputException.check(parsed != null, 'Unable to parse `before`.');
     return parsed!;
   }
+
+  /// Returns the entries from the last day.
+  ///
+  /// Keeps the [_cachedRecords] fields updated, and lists only the entries
+  /// up to the last query.
+  Future<List<AuditLogRecord>> getEntriesFromLastDay() async {
+    final now = clock.now().toUtc();
+    final cachedAge = now.difference(_cachedRecords.timestamp);
+
+    // fast track for requests within 5 seconds:
+    if (cachedAge.inSeconds < 5) {
+      return _cachedRecords.records;
+    }
+
+    // calculate window to query
+    final day = const Duration(days: 1);
+    var window = cachedAge > day ? day : (day - cachedAge);
+    if (window < Duration(minutes: 2)) {
+      window = Duration(minutes: 2);
+    }
+
+    final query = dbService.query<AuditLogRecord>()
+      ..filter('created >', now.subtract(window));
+    final current = await query.run().toList();
+
+    // merge records from cache and current query
+    final recordMap = <String, AuditLogRecord>{};
+    for (final r in _cachedRecords.records) {
+      recordMap[r.id!] = r;
+    }
+    for (final r in current) {
+      recordMap[r.id!] = r;
+    }
+    // filter records that are within the last day
+    final records = recordMap.values
+        .where((r) => now.difference(r.created!) < day)
+        .toList();
+
+    // update if current list is newer
+    if (_cachedRecords.timestamp.isBefore(now)) {
+      _cachedRecords = _CachedRecords(now, records);
+    }
+
+    return records;
+  }
```

Comment:
```suggestion
/// Returns the entries from the last day.
  ///
  /// Keeps the [_cachedRecords] fields updated, and lists only the entries
  /// up to the last query.
  Future<List<AuditLogRecord>> getEntriesFromLastDay() async {
    final now = clock.now().toUtc();
    final cachedRecords = await _cachedRecords.onError((e, _) {
      // Ignore error, we'll just not used the cached result.
      // If we don't ignore the error, then we won't be able to recover from an
      // error -- and, thus, will enter a state were all operations fail!
      return _CachedRecords(DateTime(0), []);
    });

    // fast track for requests within 5 seconds:
    if (now.difference(cachedRecords.updated) < Duration(seconds: 5)) {
      return cachedRecords.records;
    }

    _cachedRecords = Future.microtask(() async {
      var since = cachedRecords.updated;
      if (now.difference(since) > Duration(days: 1)) {
        since = now.subtract(Duration(days: 1));
      }
      // Always allow for a little clock drift
      since = since.subtract(Duration(minutes: 5));

      final query = dbService.query<AuditLogRecord>()
        ..filter('created >', since);
      final newRecords = await query.run().toList();

      // ;erge records from cachedRecords and newRecords
      final records = cachedRecords.records
          .where((r) =>
              now.difference(r.created!) < Duration(days: 1) &&
              newRecords.any((nr) => nr.id! == r.id!))
          .followedBy(newRecords)
          .toList(growable: false);

      return _CachedRecords(now, List.unmodifiable(records));
    });

    return (await _cachedRecords).records;
  }
```

Reply:
Implemented, but with a different code.